src/backend/cdb/motion/test/cdbsenddummypacket_test.c

```diff
@@ -148,10 +151,97 @@ create_sender_socket(sa_family_t af)
 	return sockfd;
 }
 
+static bool
+check_for_ipv6(void)
+{
+	struct ifaddrs *ifaddr_first, *ifaddr;
+
+	if (getifaddrs(&ifaddr_first) == -1)
+	{
+		printf("send dummy packet failed, getifaddrs failed: %m");
+		fail();
+	}
+
+	for (ifaddr = ifaddr_first; ifaddr != NULL; ifaddr = ifaddr->ifa_next)
+	{
+		if (ifaddr->ifa_addr == NULL)
```

Comment:
Combine all the three condition into single one to remove `continue`-s. It will be condition of IPv6 presence